dali/test/python/test_operator_water.py

```diff
@@ -0,0 +1,103 @@
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali as dali
+from nvidia.dali.backend_impl import TensorListGPU
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+import os
+import cv2
+from test_utils import check_batch
+from test_utils import compare_pipelines
+from test_utils import RandomDataIterator
+from PIL import Image         
+test_data_root = os.environ['DALI_EXTRA_PATH']
+caffe_db_folder = os.path.join(test_data_root, 'db', 'lmdb')
+
+
+
+class WaterPipeline(Pipeline):
+    def __init__(self, device, batch_size,  num_threads=1, device_id=0, num_gpus=1 ):
+        super(WaterPipeline, self).__init__(batch_size,
+                                           num_threads,
+                                           device_id)
+        self.device = device
+        self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+        self.decode = ops.HostDecoder(device = "cpu", output_type = types.RGB)
+        self.water = ops.Water(device = self.device, ampl_x=2., ampl_y=3., phase_x=0.2, phase_y=0.5, freq_x=0.06,freq_y=0.08, interp_type = dali.types.INTERP_LINEAR)
+        
+    
+    def define_graph(self):
+        inputs, labels = self.input(name="Reader")
+
+        images = self.decode(inputs)
+        if self.device == 'gpu':
+            images = images.gpu()
+        images = self.water(images)
+        return images
+
+def check_water_cpu_vs_gpu(batch_size):
+    compare_pipelines(WaterPipeline('cpu', batch_size),
+                      WaterPipeline('gpu', batch_size),
+                      batch_size=batch_size, N_iterations=10, eps=1)
+
+
+def python_water(img):
+    nh,nw=img.shape[:2]
+    phase_y=0.5
```

Comment:
I would put this values in one place as they are used in L27 as well. So we can avoid the risk of changing them in one place only in case this test is reworked.

Reply:
updated now